### PR TITLE
chore(ts): migrate mocks and VisibilityDetectingView

### PR DESCRIPTION
**Pull request by flochtililoch**

### CODE REVIEW

File: src/VisibilityDetectingView.tsx [hunk 1]
```diff
@@ -39,49 +37,56 @@ export default class VisibilityDetectingView extends PureComponent<Props> {
 
     // UIManager.measure may not exist during render-testing, which might break the
     // `view.measure` call
-    if (this.view && UIManager.measure) {
+    if (this.view && this.view.measure) {
```
Comment (flochtililoch) [inline src/VisibilityDetectingView.tsx:40]: suspicious - is this change needed?

Comment (hgray-instawork) [reply src/VisibilityDetectingView.tsx:40]: This was throwing an error saying that it always would resolve to true

**Review by hgray-instawork — APPROVED** [review]

File: src/VisibilityDetectingView.tsx (hunk 1 above)
Comment (flochtililoch) [reply src/VisibilityDetectingView.tsx:40]: I see - as the comment mentions above, the test is there for render-testing. I'll see if we can remove and mock instead the function in test env

Comment (flochtililoch) [reply src/VisibilityDetectingView.tsx:40]: in theory this won't change the behavior since we're testing on the method we're calling, so I'll leave it alone
